src/Platforms/AbstractPlatform.php

```diff
@@ -4017,6 +4017,10 @@ protected function getLikeWildcardCharacters(): string
      */
     public function columnsEqual(Column $column1, Column $column2): bool
     {
+        if ($column1->getColumnDefinition() !== null && $column2->getColumnDefinition() !== null) {
+            return $column1->getColumnDefinition() === $column2->getColumnDefinition();
```

Comment:
In case of the schema validation `$column1` would be populated by `\Doctrine\DBAL\Schema\AbstractSchemaManager::listTableColumns` and `$column` from the entity metadata.